### ISSUE TITLE
Add setting to disable shared spans (joins) in tracer. 

### ISSUE THREAD

**Issue by tylerbenson** [openzipkin/zipkin-ruby#125]
Other tracer implementations include a setting to force client and server spans to have different spanId's.  In Go the setting is [`WithSharedSpans(false)`](https://godoc.org/github.com/openzipkin/zipkin-go#WithSharedSpans) and in Java it's [`supportsJoin(false)`](https://static.javadoc.io/io.zipkin.brave/brave/5.0.0/brave/Tracing.Builder.html#supportsJoin-boolean-).

There should be something similar in ruby.

This is important because zipkin traces may be reported to non-zipkin backends that might not support the concept of joining spans.

**Comment by dp9463** [comment]
agree. I encountered a problem when use jaeger client send request to zipkin client, and the backend server is jaeger server. It will report the duplicate span ID.
